_data/reference.yaml

```diff
@@ -2,26 +2,42 @@ nav_section: "Reference"
 toc:
 - title: Reference
   path: /reference/index
-  section: 
-  - reference/aws.md
-  - reference/azure.md
-  - reference/gcp.md
-  - reference/openstack.md
-  - reference/kubernetes.md
-  - reference/cloud.md
+  section:
+  - title: AWS
+    path: /reference/aws.html
+  - title: Azure
+    path: /reference/azure.html
+  - title: GCP
+    path: /reference/gcp.html
+  - title: OpenStack
+    path: /reference/openstack.html
+  - title: Kubernetes
+    path: /reference/kubernetes.html
+  - title: Cloud Framework
+    path: /reference/cloud.html
 - title: Concepts
   path: /reference/concepts
   section:
-  - reference/programming-model.md
-  - reference/stack.md
-  - reference/project.md
-  - reference/config.md
-  - reference/javascript.md
-  - reference/python.md
-  - reference/component-tutorial.md
-  - reference/terraform.md
-  - reference/how.md
-  - reference/troubleshooting.md
+  - title: Programming Model
+    path: /reference/programming-model.html
+  - title: Stacks
+    path: /reference/stack.html
+  - title: Projects
+    path: /reference/project.html
+  - title: Configuration
+    path: /reference/config.html
+  - title: JavaScript and TypeScript
+    path: /reference/javascript.html
+  - title: Python
+    path: /reference/python.html
+  - title: "Tutorial: Pulumi Components"
+    path: /reference/component-tutorial.html
+  - title: Terraform Providers
+    path: /reference/terraform.html
+  - title: How Pulumi Works
+    path: /reference/how.html
+  - title: Troubleshooting
+    path: /reference/troubleshooting.html
```

Comment:
It'd be nice to not have to specify the title here explicitly, but the page lookup to get the title and resulting path from the `.md` causes the build to run for over 5 minutes. With these change the build goes down to ~40 seconds on my machine, so I think it's worth manually specifying the titles here.

Reply:
Believe it or not, I prefer listing them here explicitly. It's always been confusing to me that we had a mix of explicit and implicit, and this way you can easily see the full TOC labels in one place. Nice win on build time too 😊